prov/efa/src/efa_fabric.c

```diff
@@ -479,13 +492,40 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	ret = fi_read_file(get_sysfs_path(), gdr_path,
+			   gdr_support_str, MAX_GDR_SUPPORT_STRLEN-1);
+	if (ret < 0) {
```

Comment:
We should really be checking for FI_ENOENT here but that might be hard to do given how `fi_read_file` is written.

Reply:
Yes, fi_read_file return either -1 or number of bytes it reads. It does not return FI_ENOENT.

Reply:
but it also leaves errno in place.  So you can check for errno = ENOENT if that's what is important.

Reply:
Make sense, added check for ```ENOENT``` in new revision, also manually tested on non-gdr platform.